src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -485,6 +503,163 @@ private static GapicProductConfig createDummyInstance(
+      rpb.setInitialRetryDelayMillis(Durations.toMillis(rp.getInitialBackoff()));
+      rpb.setRetryDelayMultiplier(floatToDouble(rp.getBackoffMultiplier()));
+      rpb.setTotalTimeoutMillis(timeout);
+      rpb.setName(pName + "_params");
+
+      // construct retry codes from RetryPolicy
+      RetryCodesDefinitionProto.Builder rcb = RetryCodesDefinitionProto.newBuilder();
+      rcb.setName(pName + "_codes");
+      rp.getRetryableStatusCodesList()
+          .forEach(
+              code -> {
+                rcb.addRetryCodes(code.name());
+              });
+
+      InterfaceConfigProto.Builder ib;
+      for (Name name : mc.getNameList()) {
+        ib = builders.getOrDefault(name.getService(), InterfaceConfigProto.newBuilder());
+        if (!builders.containsKey(name.getService())) {
+          ib.setName(name.getService());
+          builders.put(name.getService(), ib);
+        }
+
+        // add retry codes & params to GAPIC interface config
+        addRetryConfigIfAbsent(ib, rcb, rpb);
+
+        // apply specific method config (overwrites) or apply service config to all methods (does
+        // not overwrite method-specific policies)
+        if (!Strings.isNullOrEmpty(name.getMethod())) {
+          findAndSetRetry(ib, true, name.getMethod(), rcb.getName(), rpb.getName(), timeout);
+        } else {
+          Interface protoService = protoInterfaces.get(ib.getName());
+          for (Method m : protoService.getMethods()) {
+            findAndSetRetry(ib, false, m.getSimpleName(), rcb.getName(), rpb.getName(), timeout);
+          }
+        }
+      }
+    }
+
+    return builders
+        .keySet()
+        .stream()
+        .map(
+            key -> {
+              return builders.get(key).build();
+            })
+        .collect(Collectors.toList());
+  }
+
+  /** converts a float to a double rounding to the nearest hundredth */
+  private static double floatToDouble(float f) {
+    return Math.round(f * 100) / 100.0;
+  }
+
+  /** adds the retry param and code list objects to the GAPIC interface unless already present */
+  static void addRetryConfigIfAbsent(
+      InterfaceConfigProto.Builder ib,
+      RetryCodesDefinitionProto.Builder rcb,
+      RetryParamsDefinitionProto.Builder rpb) {
+    boolean addRetryCodes = true;
+    for (RetryCodesDefinitionProto c : ib.getRetryCodesDefList()) {
+      if (c.getName().equals(rcb.getName())) {
+        addRetryCodes = false;
+      }
+    }
+
+    if (addRetryCodes) {
+      ib.addRetryCodesDef(rcb);
+    }
+
+    boolean addRetryParams = true;
+    for (RetryParamsDefinitionProto p : ib.getRetryParamsDefList()) {
+      if (p.getName().equals(rpb.getName())) {
+        addRetryParams = false;
+      }
+    }
+
+    if (addRetryParams) {
+      ib.addRetryParamsDef(rpb);
+    }
+  }
+
+  /**
+   * finds the named MethodConfigProto, creates one if absent in the GAPIC interface, and sets retry
+   * config
+   */
+  static void findAndSetRetry(
```

Comment:
private?

Reply:
According to the [java docs](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html), _no modifier_ is not very permissive. I wanted these methods to be visible for testing purposes. Making them private makes them harder to test

Reply:
You could add Guava's `@VisibleForTesting` so folks know that you did this intentionally.

Reply:
I will add the annotations! (implementation detail, it will come from `com.google.common.annotations.VisibleForTesting`)

Reply:
Done!